src/suggestion-box.ts

```diff
@@ -0,0 +1,409 @@
+            this.private_items[this.index]?.classList.remove("sgn-active");
+        }
+    }
+    windowClickEvent(event: any) {
+        let el = event.target;
+        while (el) {
+            if (el == this) return;
+            el = el.parentElement;
+        }
+        this.show_container = false;
+    }
+    private _format(str: any, obj: any, keys: string) {
+        const input = (this.input_el as HTMLInputElement)?.value;
+        if (typeof obj == "string" || typeof obj == "number") return obj;
+        else {
+            for (let key of Object.keys(obj)) {
+                let template: any;
+                if (key === keys) {
+                    template =
+                        this.highlight === "matched"
+                            ? this._highlight(obj[key], input.toLowerCase())
+                            : this._startsWithHighlight(obj[key]);
+                } else {
+                    template = obj[key];
+                }
+
+                str = str.replace(
+                    new RegExp("\\{" + key + "\\}", "gi"),
+                    template
+                );
+            }
+            return str;
+        }
+    }
+    keyDownInputEvent(e: any) {
+        const resultLength = this.result.length;
+        switch (e.key) {
+            case "ArrowUp":
+                this.index =
+                    this.index == -1
+                        ? resultLength - 1 + (this.has_more ? 1 : 0)
+                        : this.index - 1;
+
+                this._setActive(resultLength);
+                break;
+            case "ArrowDown":
+                this.index =
+                    this.index === -1 ||
+                    (this.has_more && this.index === resultLength)
+                        ? 0
+                        : (this.index + 1) %
+                          (resultLength + (this.has_more ? 1 : 0));
+                this._setActive(resultLength);
+                break;
+            case "Enter":
+                if (this.index === resultLength) {
+                    this._loadmoreEntry(e);
+                } else {
+                    this._setInputValue();
+                }
+                break;
+        }
+    }
+    private _inputEvent(e: any) {
+        this.index = -1;
+        this.input_el = e.target;
+        let min_char = Number(this.minchar);
+        if ((this.input_el as HTMLInputElement)?.value.length >= min_char) {
+            if (this.list) {
+                if ((this.input_el as HTMLInputElement)?.value) {
+                    this.filtered_list = this.list.filter((item) =>
+                        item[this.listkey].includes(
+                            (this.input_el as HTMLInputElement)?.value
+                        )
+                    );
+                    this.setSuggestion(this.filtered_list, false, this.listkey);
+                } else {
+                    this.show_empty = true;
```

Comment:
Even if empty is true, the show_container could be false here right? As per your code, you setting show_container to false when window click event occurs. Lets say you have typed something, suggestions are shown and then you click outside the dropdown. Now click on the input box to make it gain focus, at this the dropdown should be shown. But as per this logic, the dropdown will not be shown.